system/security/shared/caching.cpp

```diff
@@ -672,16 +680,55 @@ bool CPermissionsCache::queryPermsManagedFileScope(ISecUser& sec_user, const cha
 
 SecAccessFlags CPermissionsCache::queryDefaultPermission(ISecUser& user)
 {
-    if (m_defaultPermission == SecAccess_Unknown)
+
+    if (m_useLegacyDefaultFileScopePermissionCache)
     {
-        if (m_secMgr)
-            m_defaultPermission = m_secMgr->queryDefaultPermission(user);
+        if (m_defaultPermission == SecAccess_Unknown)
+        {
+            if (m_secMgr)
+                m_defaultPermission = m_secMgr->queryDefaultPermission(user);
+            else
+                m_defaultPermission = SecAccess_Full;
+
+            DBGLOG("Legacy default file scope permission set to %s(%d) for all users, based on User '%s'", getSecAccessFlagName(m_defaultPermission),
+                   m_defaultPermission, user.getName());
+        }
+        return m_defaultPermission;
+    }
+
+    SecAccessFlags defaultPermission = SecAccess_None;
+    if (m_secMgr)
+    {
+        const std::string username(user.getName());
+        ReadLockBlock defaultScopesReadLock(m_defaultScopesRWLock);
+        auto it = m_userDefaultFileScopePermissions.find(username);
+        if (it == m_userDefaultFileScopePermissions.end())
+        {
+            defaultScopesReadLock.clear();
+            WriteLockBlock defaultScopesWriteLock(m_defaultScopesRWLock);
+
+            auto verifyIt = m_userDefaultFileScopePermissions.find(username);
+            if (verifyIt == m_userDefaultFileScopePermissions.end())
+            {
+                defaultPermission = m_secMgr->queryDefaultPermission(user);
+                m_userDefaultFileScopePermissions.emplace(username, defaultPermission);
+                DBGLOG("Added user '%s' to default file scope permissions with access %s(%d)", username.c_str(), getSecAccessFlagName(defaultPermission),
```

Comment:
Should this logging be protected with a flag e.g. traceLDAP?

Reply:
As we move towards more secure permissions and removal of Authenticated users from default file scope permissions, I think this information is useful in helping lock down a cluster. In fact, I debated making it a PROGLOG so that the information on default scopes assigned to users is available. If you think a flag is more appropriate, I can add one. I think now is the time if we are going to do it.

Reply:
This will be logged for the first lookup for each user every time the cache is cleared.  Given the potential cost of logging that might add up.  Keep as is for the moment, but review how much logging it is generating once this is being used on a live system.
(Simple support for trace flags is not yet available in esp, when it is it will be much simpler to add a guard.)